sdk/core/src/services/WalletStorageService.ts

```diff
@@ -371,15 +365,137 @@ export default class WalletStorageService {
+        credentialTypes: credential.credentialTypes,
+        payload: credential.payload,
+      })
+
+      responses.push(body)
+    }
+
+    return responses
+  }
+
+  public async getAllCredentials(audienceDid: string, types: string[][], storageRegion?: string) {
+    const token = await this._authorizeVcVault(audienceDid)
+    storageRegion = storageRegion || this._storageRegion
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+
+    // convert each type to hash
+    const encryptedTypes: string[][] = []
+    for (const subset of types) {
+      const encryptedSubset: string[] = []
+      for (const type of subset) {
+        const encryptedType = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, type)
+        encryptedSubset.push(encryptedType)
+      }
+      encryptedTypes.push(encryptedSubset)
+    }
+
+    const { body } = await this._vaultApiService.searchCredentials(token, storageRegion, encryptedTypes)
+    const credentials = await this._decryptCredentials(body.credentials)
+
+    // should be deleted during migration to affinidi-vault Phase #2
+    const bloomCredentials = await this.fetchAllDecryptedCredentials()
+    if (types.length === 0) {
+      credentials.push(...bloomCredentials)
+      return credentials
+    }
+
+    for (const credential of bloomCredentials) {
+      const credentialTypes = Array.isArray(credential?.type) ? (credential.type as string[]) : []
+
+      const isFits = types.some((subtypes) => subtypes.every((subtype) => credentialTypes.includes(subtype)))
+
+      if (isFits) {
+        credentials.push(credential)
+      }
+    }
+
+    return credentials
+  }
+
+  public async getCredentialById(audienceDid: string, credentialId: string, storageRegion?: string): Promise<any> {
+    const token = await this._authorizeVcVault(audienceDid)
+    storageRegion = storageRegion || this._storageRegion
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+
+    const hashedId = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, credentialId)
+
+    let encryptedCredential: string
+    try {
+      const { body, status } = await this._vaultApiService.getCredential(token, storageRegion, hashedId)
+      if (status >= 400) {
+        throw new SdkErrorFromCode('COR-23', { credentialId })
+      }
+
+      encryptedCredential = body.payload
+    } catch (error) {
+      // should be deleted during migration to affinidi-vault Phase #2
+      const bloomToken = await this.authorizeVcBloomVault()
+      const credentialIndexToDelete = await this._findCredentialIndexById(credentialId)
+      const { body, status } = await this._bloomVaultApiService.getCredentials({
+        accessToken: bloomToken,
+        start: credentialIndexToDelete,
+        end: credentialIndexToDelete,
+        storageRegion,
       })
+      if (status >= 400 && body?.length > 0) {
```

Comment:
`GenericApiService` throws an error for any status that does not start with `2`. So this condition will never be satisfied (and so is the one above for `AffinidiVault`).

Reply:
hm, that means we should put calls to bloom-vault into try catch as well, otherwise we will be throwing error from bloom-vault in case both encountered an error